python/pyspark/sql/connect/session.py

```diff
@@ -93,14 +94,13 @@
     from pyspark.sql.connect.udtf import UDTFRegistration
 
 
-# `_active_spark_session` stores the active spark connect session created by
-# `SparkSession.builder.getOrCreate`. It is used by ML code.
-#  If sessions are created with `SparkSession.builder.create`, it stores
-#  The last created session
-_active_spark_session = None
-
-
 class SparkSession:
+    # The active SparkSession for the current thread
+    _active_session = threading.local()
```

Comment:
Need the type hint for this?

Reply:
I am actually not sure how to type this .. it's a private var so should be fine.

Reply:
I guess `ClassVar[threading.local]`?
But ok, should be fine.